samcli/hook_packages/terraform/hooks/prepare/resource_linking.py

```diff
@@ -207,6 +207,12 @@ def _build_module_resources_from_configuration(module_configuration: Dict, modul
 
         expressions = config_resource.get("expressions", {})
         for expression_name, expression_value in expressions.items():
+            # we do not process the attributes of type dictionary
+            # Todo add dictionary type attributes post beta
+            if isinstance(expression_value, list):
+                LOG.debug("Skip processing the attribute %s as its value is a map.", expression_name)
+                continue
+
```

Comment:
Found a use case that is not supported in this code. If the attribute value is a block (dictionary) .. it got translated as a list of dictionaries, and each key in the attribute value dictionary will be an item in this list. 

I skipped processing this attribute for now as it is not affecting this beta version, as no of the attributes that we care about for beta 1 stories has a block value